api/src/pcapi/core/offerers/models.py

```diff
@@ -1095,6 +1096,13 @@ def is_caledonian(self) -> bool:
             return True
         return False
 
+    @is_caledonian.expression  # type: ignore[no-redef]
+    def is_caledonian(cls) -> BinaryExpression:  # pylint: disable=no-self-argument
+        return sa.or_(
+            cls.siren.ilike("NC%"),
+            cls.postalCode.ilike("988%"),
```

Comment:
Détail : Pas besoin de _case-insensitive_, des `like` suffisent.
Et c'est mieux d'utiliser les variables, d'où ma suggestion : 
```suggestion
            cls.siren.ilike(f"{siren_utils.NEW_CALEDONIA_SIREN_PREFIX}%"),
            cls.postalCode.ilike(f"{region_utils.NEW_CALEDONIA_DEPARTMENT_CODE}%"),
```